src/plugins/vis_type_timeseries/public/request_handler.ts

```diff
@@ -29,39 +29,55 @@ interface MetricsRequestHandlerParams {
   input: KibanaContext | null;
   uiState: Record<string, any>;
   visParams: TimeseriesVisParams;
+  searchSessionId?: string;
```

Comment:
nit: data plugin exports `ISearchOptions`.  Maybe we should map to ISearchOptions['sessionId'] instead of passing optional string. From my point of view, this is more correct

Reply:
Hmm, in my opinion, it is an overkill to look at it this way. If I understand correctly, following this recommendation would require updating the type for all places where `searchSessionId` is passed. This is a pretty significant diff which I think doesn't worth it. 

The same would apply to any other place we type stuff as ambiguous strings. for example
Instead `indexPatternId: string`, use `indexPatternId: IndexPattern['id']`

Reply:
up to you, but I prefer to write `indexPatternId: IndexPattern['id'] ` =)